server/game/gamesteps/uiprompt.js

```diff
@@ -87,6 +98,13 @@ class UiPrompt extends BaseStep {
      */
     onCompleted() {
     }
+
+    /**
+     * Will be implemented in sub classes that have a specific player that is using the prompt
+     */
+    getPlayer() {
```

Comment:
I do notice that `AllPlayerPrompt`, which inherits from this class, does not implement this method. It's used for at least the plot selection prompt, which seems like it should be subject to timers. Any thoughts on making it compatible with the timer?

(you could have this method return an array of players that are currently making choices instead, for instance)

Reply:
Good point regarding the AllPlayerPrompt and its sub-classes. I made a re-test to make sure the plot phase was working "as expected" as in when one player confirms his/her plot, the timer stops. It does work like that at the moment, but the reason for that is not the stopClock call in the completed() method of uiprompt but the stopClock call in the setPrompt() method for the waitingPrompt case. 
After figuring that out, i questioned the necessity of the stopClock call in completed() and the whole getPlayer() stuff. Testing that again, i still need that stopClock in completed for situations like initiating a challenge and having the first action in the after-attacker-declared action window. Without the stopping in completed, the attacking players clock would only stop after it´s the other players turn for taking actions in the after-attackers-declared window.
So all in all, it is not 100% necessary to have the getPlayer implemented in the allPlayerPrompt, but it won´t hurt to have it there. I will follow your advice and return an array of players and stop the clock for all players in the array